smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolUnitTestErrorGenerator.kt

```diff
@@ -76,7 +83,25 @@ open class HttpProtocolUnitTestErrorGenerator protected constructor(builder: Bui
 
     override fun renderExpectedBody(test: HttpResponseTestCase) {
         if (test.body.isPresent && test.body.get().isNotBlank()) {
-            val data = writer.format("Data(\"\"\"\n\$L\n\"\"\".utf8)", test.body.get().replace("\\\"", "\\\\\""))
+            val isCbor = ctx.service.requestWireProtocol == WireProtocol.CBOR ||
+                ctx.service.awsProtocol == AWSProtocol.RPCV2_CBOR ||
+                ctx.service.responseWireProtocol == WireProtocol.CBOR ||
+                test.protocol == Rpcv2CborTrait.ID
+
+            val bodyContent = test.body.get().replace("\\\"", "\\\\\"")
+            val data: String = if (isCbor) {
+                // Attempt to decode Base64 data once for CBOR
+                try {
```

Comment:
CBOR data is binary data but for protocol tests it need to be passed as a base64 encoded string